### ISSUE TITLE
how to get the "tree index"  when when training model.fit

### ISSUE THREAD

**Issue by xu-peng-7** [catboost/catboost#1156]
Problem: I want to add my custom_loss in different  " tree index " when training model.fit  .But i can't  get the "tree index" in my program. And How can i customize the format of  training stdout ? ths.
catboost version:
Operating System: windows 10
CPU:
GPU:

**Comment by annaveronika** [comment]
It is not possible to customize stdout. What do you mean by tree index? Do you want to calculate loss every k-th iteration?

**Comment by xu-peng-7** [comment]
> It is not possible to customize stdout. What do you mean by tree index? Do you want to calculate loss every k-th iteration?

I want to adjust the loss function according to different iterations ，For example, the 1-th  loss will plus  extra 1 ，the 2-th  loss will plus extra 2 .....  But I don't know the current k-th when running model.fit

**Comment by annaveronika** [comment]
No, it's not possible to use number of iteration in a custom loss function unfortunatelly

**Comment by xu-peng-7** [comment]
> No, it's not possible to use number of iteration in a custom loss function unfortunatelly

ok~ thanks for your help.

**Comment by annaveronika** [comment]
What you can do, is you can use staged_predict, calculate formula values on every iteration and then calculate metric on top of that.
